Dynatrace/dynatrace-operator
# Removed duplicated oneagent clusterrole, clusterrolebinding and serviceaccount

# Description

`oneagent-privileged` and `oneagent-unprivileged` ClusterRoles are exactly the same so the duplicate is removed.
Related ClusterRoleBindings and ServiceAccounts are modified accordingly. 

`TestPodSpecServiceAccountName` test is updated only (not removed) to make sure single ServiceName is used from now on.

## How can this be tested?
Deploy CloudNative DTO on kubernetes cluster and openshift cluster with enabled and disabled `feature.dynatrace.com/oneagent-privileged` feature flag. Make sure OneAgent PODs are created and works properly.

## Checklist
